### PR TITLE
fix(ci): don't report a broken fragment if duplicate fragment exists

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- addresses https://github.com/mdn/content/pull/35441#issuecomment-2306079810

If a file has two headers with the same text and one header is updated, then the script reports it as a broken fragment. The PR adds logic to check if the file has headers with the same content.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@pepelsbey Can you please look at this sooner rather than later? I think it would be good for this to go in before I merge  #35441, but I also would like that in for the release this week.

File: scripts/log-url-issues.js [hunk 1]
```diff
@@ -97,7 +98,25 @@ function getFragmentDetails(fromStaging = true) {
         .map((header) => header.replace(/-#+ /g, ""))
         .map((header) => stringToFragment(header))
         .filter((header) => !addedFragments.includes(header))
-        .forEach((header) => deletedFragmentDetails.push(`${path}#${header}`));
+        .filter((header) => {
+          // check if another header with same name exists in the file
+          const content = fs.readFileSync(
```
[the remark this answers is not in the document]
Comment (OnkarRuikar) [reply scripts/log-url-issues.js]: I've memorized it. What does "file reads out of band" mean?

**Comment by bsmth** [comment]
> @pepelsbey Can you please look at this sooner rather than later? I think it would be good for this to go in before I merge #35441, but I also would like that in for the release this week.

Vadim is on PTO this week so I've added myself as a reviewer

**Review by Josh-Cena — APPROVED** [review]
Tested and it works. Letting @bsmth take another look. Otherwise I'll merge it tonight, just so @hamishwillee can work on his stuff when he wakes up.

**Review by bsmth — APPROVED** [review]
Thanks, all, let's give it a spin 👍🏻

**Pull request merged**